store/src/java/com/zimbra/cs/imap/ImapServerListener.java

```diff
@@ -185,11 +234,21 @@ private void initWaitSet(String accountId, boolean alreadyListening) throws Serv
                 waitSetReq.addAddAccount(updateOrAdd);
             }
         }
-        ZimbraLog.imap.debug("Sending initial AdminWaitSetRequest. WaitSet ID: %s", wsID);
-        AdminWaitSetResponse wsResp = soapProv.invokeJaxbAsAdminWithRetry(waitSetReq, server);
+
         try {
+            ZimbraLog.imap.debug("Sending initial AdminWaitSetRequest. WaitSet ID: %s", wsID);
+            AdminWaitSetResponse wsResp = soapProv.invokeJaxbAsAdminWithRetry(waitSetReq, server);
             processAdminWaitSetResponse(wsResp);
-            continueWaitSet();
+        } catch (SoapFaultException e) {
+            if(AdminServiceException.NO_SUCH_WAITSET.equalsIgnoreCase(e.getCode())) {
+                //waitset is gone. Create a new one
+                ZimbraLog.imap.warn("AdminWaitSet %s does not exist anymore", wsID);
```

Comment:
NO_SUCH_WAITSET is not expected under normal conditions. It will happen if remote mailboxd was restarted or if a sweeper on remote mailboxd killed this waitset. The latter may  indicate that some configuration settings should be updated.